uberfire-services/uberfire-services-backend/src/main/java/org/guvnor/common/services/backend/test/TestRunnerServiceImpl.java

```diff
@@ -60,18 +60,43 @@ public void runAllTests(final String identifier,
     public void runAllTests(final String identifier,
                             final Path path,
                             final Event<TestResultMessage> customTestResultEvent) {
-
-        final TestRunEventCollection testResultEvent = new TestRunEventCollection();
+        final ArrayList<TestRunEventCollection> events = new ArrayList<>();
 
         for (final TestService testService : testServices) {
+            final TestRunEventCollection testResultEvent = new TestRunEventCollection();
```

Comment:
@Rikkola 
HI, I'm reading on browser, so maybe misreading. 
Anyway, it seems to me that inside loop at line 65 you populate a collection, then with methods called at lines 73-75 you iterate the same collection three times (one for method) to calculate the parameter used inside the fire event:  would not be possible to do such calculation directly inside the loop at line 65 ? Not sure it is worth anything, maybe just if the input is big

Reply:
The collection size will probably never go above 5, but better to be safe just sum them up on the go. I'll do change.

Reply:
Fixed.